ui/app/services/system.js

```diff
@@ -109,17 +110,95 @@ export default class SystemService extends Service {
     });
   }
 
-  @computed('token.selfToken')
+  @computed('defaults', 'token.selfToken', 'variableDefaults.Region')
   get defaultRegion() {
-    const token = this.token;
-    return PromiseObject.create({
-      promise: token
-        .authorizedRawRequest(`/${namespace}/agent/members`)
-        .then(jsonWithDefault({}))
-        .then((json) => {
-          return { region: json.ServerRegion };
-        }),
+    // // let variableDefaultRegion = this.variableDefaults.Region;
+    // let variableDefaultRegion = false;
+    // if (variableDefaultRegion) {
+    //   return { region: variableDefaultRegion };
+    return this.defaults.then((defaults) => {
+      if (defaults.region) {
+        return { region: defaults.region };
+      } else {
+        const token = this.token;
+        return PromiseObject.create({
+          promise: token
+            .authorizedRawRequest(`/${namespace}/agent/members`)
+            .then(jsonWithDefault({}))
+            .then((json) => {
+              return { region: json.ServerRegion };
+            }),
+        });
+      }
     });
+    // if (this.defaults.region) {
+    //   return { region: this.defaults.region };
+    // } else {
+    //   const token = this.token;
+    //   return PromiseObject.create({
+    //     promise: token
+    //       .authorizedRawRequest(`/${namespace}/agent/members`)
+    //       .then(jsonWithDefault({}))
+    //       .then((json) => {
+    //         return { region: json.ServerRegion };
+    //       }),
+    //   });
+    // }
+  }
+
+  defaultProperties = [
+    'userDefaultRegion',
+    'userDefaultNamespace',
+    'userDefaultNodePool',
+  ];
+
+  async establishUIDefaults() {
+    // // First, check to see if there are localStorage properties set for each of the defaults.
+    // // If there are, we don't need to reach out to check the variable or agent config.
+    // if (this.defaultProperties.every((defaultString) => this[defaultString])) {
+    //   return;
+    // }
+
+    let agent = await this.agent;
+    this.agentDefaults = agent?.config?.UI?.Defaults || {};
+    let variableDefaults = await this.fetchVariableDefaults();
+    this.variableDefaults = variableDefaults || {};
+    return {
+      agentDefaults: this.agentDefaults,
+      variableDefaults: this.variableDefaults,
+    };
+  }
+
+  async fetchVariableDefaults() {
+    try {
+      // if (this.can.can('read variable', 'nomad/ui/defaults', '*')) { // TODO: is wildcard correctly handled by "can"?
```

Comment:
Tested without `can` ability checks and this fails gracefully. Can probably remove the condition.